### ISSUE TITLE
Can't build Docker image

### ISSUE THREAD

**Issue by OrigamiDream** [huggingface/text-generation-inference#763]
### System Info

Host computer spec:

`nvidia-smi`
- Driver Version: 525.105.17
- CUDA Version: 12.0
- NVIDIA GeForce RTX3080 10GiB x4

`uname -a`
- Linux ubuntu 5.19.0-45-generic # 46~22.04.1-Ubuntu SMP PREEMPT_DYNAMIC Wed Jun 7 15:06:04 UTC 20 x86_64 x86_64 x86_64 GNU/Linux

### Information

- [X] Docker
- [ ] The CLI directly

### Tasks

- [X] An officially supported command
- [ ] My own modifications

### Reproduction

1. git clone https://github.com/huggingface/text-generation-inference.git
2. cd text-generation-inference.git
3. Prune all Docker images prior to do following jobs.
4. docker build -t huggingface/text-generation-inference:latest .
5. The error happens on `flash-att-builder` stage

```
running build_ext
Traceback (most recent call last):
  File "/usr/src/flash-attention/setup.py", line 175, in <module>
    setup(
  File "/opt/conda/lib/python3.9/site-packages/setuptools/__init__.py", line 87, in setup
    return distutils.core.setup(**attrs)
  File "/opt/conda/lib/python3.9/site-packages/setuptools/_distutils/core.py", line 185, in setup
    return run_commands(dist)
  File "/opt/conda/lib/python3.9/site-packages/setuptools/_distutils/core.py", line 201, in run_commands
    dist.run_commands()
  File "/opt/conda/lib/python3.9/site-packages/setuptools/_distutils/dist.py", line 969, in run_commands
    self.run_command(cmd)
  File "/opt/conda/lib/python3.9/site-packages/setuptools/dist.py", line 1208, in run_command
    super().run_command(command)
  File "/opt/conda/lib/python3.9/site-packages/setuptools/_distutils/dist.py", line 988, in run_command
    cmd_obj.run()
  File "/opt/conda/lib/python3.9/site-packages/setuptools/_distutils/command/build.py", line 132, in run
    self.run_command(cmd_name)
  File "/opt/conda/lib/python3.9/site-packages/setuptools/_distutils/cmd.py", line 318, in run_command
    self.distribution.run_command(command)
  File "/opt/conda/lib/python3.9/site-packages/setuptools/dist.py", line 1208, in run_command
    super().run_command(command)
  File "/opt/conda/lib/python3.9/site-packages/setuptools/_distutils/dist.py", line 988, in run_command
    cmd_obj.run()
  File "/opt/conda/lib/python3.9/site-packages/setuptools/command/build_ext.py", line 84, in run
    _build_ext.run(self)
  File "/opt/conda/lib/python3.9/site-packages/setuptools/_distutils/command/build_ext.py", line 346, in run
    self.build_extensions()
  File "/opt/conda/lib/python3.9/site-packages/torch/utils/cpp_extension.py", line 484, in build_extensions
    _check_cuda_version(compiler_name, compiler_version)
  File "/opt/conda/lib/python3.9/site-packages/torch/utils/cpp_extension.py", line 368, in _check_cuda_version
    torch_cuda_version = packaging.version.parse(torch.version.cuda)
  File "/opt/conda/lib/python3.9/site-packages/pkg_resources/_vendor/packaging/version.py", line 49, in parse
    return Version(version)
  File "/opt/conda/lib/python3.9/site-packages/pkg_resources/_vendor/packaging/version.py", line 264, in __init__
    match = self._regex.search(version)
TypeError: expected string or bytes-like object
make: *** [Makefile:10: build-flash-attention] Error 1
The command '/bin/sh -c make build-flash-attention' returned a non-zero code: 2
```

I think the `pytorch-install` stage does not install CUDA properly.
When Docker runs `cd flash-attention && python setup.py build`, following warning has occurred:
```
Turn off this advice by setting config variable advice.detachedHead to false

HEAD is now at 3a9bfd0 [FT] rotary_cos/sin should have shape (dim) instead of (seqlen, dim)
cd flash-attention && python setup.py build
No CUDA runtime is found, using CUDA_HOME='/opt/conda'
Submodule 'csrc/flash_attn/cutlass' (https://github.com/NVIDIA/cutlass.git) registered for path 'csrc/flash_attn/cutlass'
Cloning into '/usr/src/flash-attention/csrc/flash_attn/cutlass'...
Submodule path 'csrc/flash_attn/cutlass': checked out '319a389f42b776fae5701afcb943fc03be5b5c25'
```

### Expected behavior

The Docker image must be built successfully.

**Comment by mael-le-gal** [comment]
I got the same issue yesterday. It appears that the conda pytorch 2.0.0 requirement described in the Dockerfile doesn't resolve anymore to the "cuda version" of pytorch. I don't understand exactly why.

On my side upgrading the `ARG PYTORCH_VERSION=2.0.0` to `ARG PYTORCH_VERSION=2.0.1` in the Dockerfile solved the issue.

**Comment by OrigamiDream** [comment]
In my opinion, It seems PyTorch 2.0.0 index repo has been changed to mirror cu117 as default instead of cu118. (No verified)

**Comment by Narsil** [comment]
Should be fixed in [sha-f91e9d2](https://github.com/orgs/huggingface/packages/container/text-generation-inference/115569670?tag=sha-f91e9d2)

**Comment by OrigamiDream** [comment]
The issue seems to be resolved. Thanks!
